packages/vm/src/crypto.rs

```diff
@@ -1,31 +1,53 @@
+use ed25519_zebra as ed25519;
+
 use k256::{
     ecdsa::signature::{DigestVerifier, Signature as _}, // traits
     ecdsa::{Signature, VerifyingKey},                   // type aliases
 };
 use sha2::Digest; // trait
 
+use std::convert::{TryFrom, TryInto};
+
 use crate::dummy_sha2;
 use crate::errors::{VmError, VmResult};
 
-pub fn secp256k1_verify(message_hash: &[u8], signature: &[u8], public_key: &[u8]) -> VmResult<()> {
+pub fn secp256k1_verify(
+    message_hash: &[u8],
+    signature_bytes: &[u8],
+    public_key_bytes: &[u8],
+) -> VmResult<()> {
     // Already hashed, just build Digest container
     let message_digest = dummy_sha2::Sha256::new().chain(message_hash);
 
     let mut signature =
-        Signature::from_bytes(signature).map_err(|e| VmError::crypto_err(e.to_string()))?;
+        Signature::from_bytes(signature_bytes).map_err(|e| VmError::crypto_err(e.to_string()))?;
     // Non low-S signatures require normalization
     signature
         .normalize_s()
         .map_err(|e| VmError::crypto_err(e.to_string()))?;
 
-    let public_key = VerifyingKey::from_sec1_bytes(public_key)
+    let public_key = VerifyingKey::from_sec1_bytes(public_key_bytes)
         .map_err(|e| VmError::crypto_err(e.to_string()))?;
 
     public_key
         .verify_digest(message_digest, &signature)
         .map_err(|e| VmError::crypto_err(e.to_string()))
 }
 
+pub fn ed25519_verify(
+    message_hash: &[u8],
+    signature_bytes: &[u8],
+    public_key_bytes: &[u8],
+) -> VmResult<()> {
+    // Deserialize
+    let res: Result<ed25519::Signature, ed25519::Error> = signature_bytes.try_into();
```

Comment:
```suggestion
    let res = ed25519::Signature::try_from(signature_bytes);
```

would be simpler I guess